include/deal.II/numerics/nonlinear.h

```diff
@@ -0,0 +1,401 @@
+       */
+      enum SolutionStrategy
+      {
+        /**
+         * Standard Newton iteration.
+         */
+        newton = KIN_NONE,
+        /**
+         * Newton iteration with linesearch.
+         */
+        linesearch = KIN_LINESEARCH,
+        /**
+         * Fixed point iteration.
+         */
+        fixed_point = KIN_FP,
+        /**
+         * Picard iteration.
+         */
+        picard = KIN_PICARD,
+      };
+
+      AdditionalData(const std::string &solver_type = "kinsol",
+                     const SolutionStrategy &strategy = linesearch,
+                     const unsigned int maximum_non_linear_iterations   = 200,
+                     const double       function_tolerance              = 0.0,
+                     const double       step_tolerance                  = 0.0,
+                     const bool         no_init_setup                   = false,
+                     const unsigned int maximum_setup_calls             = 0,
+                     const double       maximum_newton_step             = 0.0,
+                     const double       dq_relative_error               = 0.0,
+                     const unsigned int maximum_beta_failures           = 0,
+                     const unsigned int anderson_subspace_size          = 0,
+                     const double       absolute_tolerance              = 1.e-20,
+                     const double       relative_tolerance              = 1.e-5,
+                     const unsigned int threshold_nonlinear_iterations  = 1,
+                     const unsigned int threshold_n_linear_iterations   = 0,
+                     const bool         reuse_solver                    = false);
+
+      // void
+      // add_parameters(ParameterHandler &prm);
+
+      //SolverType solver_type;
+      std::string solver_type;
+      SolutionStrategy strategy;
+
+      unsigned int maximum_non_linear_iterations;
+      double function_tolerance;
+      double step_tolerance;
+      bool no_init_setup;
+      unsigned int maximum_setup_calls;
+      double maximum_newton_step;
+      double dq_relative_error;
+      unsigned int maximum_beta_failures;
+      unsigned int anderson_subspace_size;
+      const double absolute_tolerance;
+      const double relative_tolerance;
+      const unsigned int threshold_nonlinear_iterations;
+      const unsigned int threshold_n_linear_iterations;
+      const bool reuse_solver;
+    };
+
+  NLSolverSelector();
+
+  NLSolverSelector(const AdditionalData &additional_data);
+
+  // Trying to delegate a constructor
+  NLSolverSelector(const AdditionalData &additional_data,
+                   const Teuchos::RCP<Teuchos::ParameterList> &parameters);
+
+  //~NLSolverSelector();
+
+  void
+  select(const std::string &type);
+
+
+  void 
+  set_data(const typename TrilinosWrappers::NOXSolver<VectorType>::AdditionalData &additional_data,
+           const Teuchos::RCP<Teuchos::ParameterList> &parameters);
```

Comment:
Add a default for ``parameters``

Reply:
Is that not what [this](https://github.com/tjhei/dealii/pull/18/files/bcf69b2e6eeaba10241020b36df212d0b3e084e9#diff-0d3c62ae924f7a85917350aadaf7161cdc3280d916047590e485740ecc3b76aaR174) does already?